code/graphics/2d.cpp

```diff
@@ -137,17 +138,55 @@ static bool gamma_change_listener(float new_val, bool initial)
 	return true;
 }
 
+static void parse_gamma_func()
+{
+	float value;
+	stuff_float(&value);
+
+	constexpr float EPSILON = 0.0001f;
+
+	if (value < 0.1f - EPSILON || value > 5.0f + EPSILON) {
+		error_display(0, "%f is not a valid gamma value! (Out of range)", value);
+		return;
+	}
+
+	float expected_i = value / 0.05f;
+	int i = fl2i(std::round(expected_i));
+
+	if (std::abs(value - (0.05f * i)) < EPSILON && i >= 2 && i <= 100) {
+		Gr_gamma = value;
+		return;
+	}
+
+	error_display(0, "%f is not a valid gamma value! (Invalid increment)", value);
+}
+
 static auto GammaOption __UNUSED = options::OptionBuilder<float>("Graphics.Gamma",
                      std::pair<const char*, int>{"Brightness", 1375},
                      std::pair<const char*, int>{"The brightness value used for the game window", 1738})
                      .category(std::make_pair("Graphics", 1825))
-                     .default_val(1.0f)
+                     .default_func([]() { return Gr_gamma; })
                      .enumerator(gamma_value_enumerator)
                      .display(gamma_display)
                      .change_listener(gamma_change_listener)
                      .flags({options::OptionFlags::RetailBuiltinOption})
+                     .parser(parse_gamma_func)
                      .finish();
 
+static void parse_lighting_func()
+{
+	int value[static_cast<int>(DefaultDetailLevel::Num_detail_levels)];
+	stuff_int_list(value, static_cast<int>(DefaultDetailLevel::Num_detail_levels), RAW_INTEGER_TYPE);
+
+	for (int i = 0; i < static_cast<int>(DefaultDetailLevel::Num_detail_levels); i++) {
+
+		if (value[i] < 0 || value[i] > static_cast<int>(DefaultDetailLevel::Num_detail_levels)) {
```

Comment:
In `mod_table.cpp` there is the nearly same detail level check, though in that function the if condition is `if (detail_level < 0 || detail_level > static_cast<int>(DefaultDetailLevel::Num_detail_levels) - 1) {` with specifically `Num_detail_levels) - 1`.

I reckon this check should also use `Num_detail_levels) - 1` since `Num_detail_levels = 4` but the valid values are 0-3. :)

Reply:
As discussed in DMs, this revealed that the language between levels and values still isn't clear so I've now clarified between presets, levels, and values in a new commit. That commit also resolves this issue by getting the `4` from the correct define.